### PR TITLE
Typescript error busting - contributionsLandingReducer.ts

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?

Fix the errors in `contributionsLandingReducer.ts`. This mainly involved types around Amazon Pay. I moved all the the Amazon Pay types to their own file, to avoid circular imports. I then cleaned up parts of the redux state that were unused. 

[**Trello Card**](https://trello.com/c/KRylYdzA/302-typescript-error-busting-contributionslandingreducerts)

### CODE REVIEW

File: support-frontend/assets/helpers/forms/paymentIntegrations/amazonPay/types.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// ---- Types ---- //
+
+export interface AmazonLoginObject {
```
Comment (TomPretty) [inline support-frontend/assets/helpers/forms/paymentIntegrations/amazonPay/types.ts:3]: This is the new file containing all of the types for Amazon Pay

File: support-frontend/assets/helpers/forms/paymentIntegrations/amazonPay/setup.ts [hunk 2]
```diff
@@ -7,30 +7,27 @@ import {
 	setAmazonPayPaymentsObject,
 } from 'pages/contributions-landing/contributionsLandingActions';
 
-const amazonPaySupportedCountryGroups = ['UnitedStates'];
+// ---- Setup ---- //
```
Comment (TomPretty) [inline support-frontend/assets/helpers/forms/paymentIntegrations/amazonPay/setup.ts:11]: The refactors here are just moving things around

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 3]
```diff
@@ -59,88 +66,18 @@ export interface StripeCardFormData {
 	createPaymentMethod: ((clientSecret: string | null) => void) | null;
 	handle3DS: ((clientSecret: string) => Promise<PaymentIntentResult>) | null; // For single only
 }
-interface Wallet {
-	bind: (ID: string) => void;
-}
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:64]: All of these types have been moved to the new file

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 4]
```diff
@@ -188,7 +128,6 @@ function createFormReducer() {
 		thirdPartyPaymentLibraries: {
 			ONE_OFF: {
 				Stripe: null,
-				AmazonPay: null,
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:191]: this was unused

File: support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts [hunk 5]
```diff
@@ -203,7 +142,6 @@ function createFormReducer() {
 				amazonLoginObject: null,
 				amazonPaymentsObject: null,
 			},
-			loginButtonReady: false,
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingReducer.ts:206]: this was unused

**Review by rupertbates — APPROVED** [review]
👍 looks good, just one suggestion

**Pull request merged**